### ISSUE TITLE
project-config sync/rebuild errors with missing fields

### ISSUE THREAD

**Issue by timkelty** [craftcms/cms#5209]
Following a Craft 2 upgrade, I'm able to run `craft project-config/rebuild`, which succeeds.

There are certain fieldtypes that have no match – they remain in `project.yaml`, and show red in the CP.

I would assume, that running `craft project-config/rebuild` immediately following `rebuild` would always succeed, but in this case it doesn't. I get errors about missing field classes for the fields that no longer have types.

**Comment by andris-sevcenko** [comment]
> I would assume, that running `craft project-config/rebuild` immediately following `rebuild` would

Do you mean "running `craft project-config/force`" here?

`rebuild` populates whatever your actual state of Craft is to project config. If you have fields that use classes that are not available to them, I would not expect things to run smoothly.  How should Craft proceed about resolving this, then?

**Comment by timkelty** [comment]
> Do you mean "running craft project-config/force" here?

Yup, mis-typed, (as I think you did 😁 )…meaning `craft project-config sync --force`.

I think a reasonable expectation is that Craft itself will never build an invalid config. In this case that's what's happening, as the config it generates cannot be sync'd.

Not sure the best approach, but Craft could simple skip over fields that had missing classes, rather than halting on exception.

**Comment by andris-sevcenko** [comment]
> Yup, mis-typed, (as I think you did 😁)

So it would seem :)

Well, project config rebuild operation essentially re-runs the migration that creates the initial project config, so you could argue that the config after upgrading isn't syncable.

As this scenario seems pretty fringe and unlikely to occur during normal operation, if I was upgrading a site, I would prefer it breaking with a screaming exception rather than just silently ignoring the errors. Furthermore, if something in the project config gets glossed over, it still gets imported into the internal project config, meaning that you would have to use `--force` option next time to try to sync it again.
